diff --git a/nexfetch b/nexfetch
index 48b96d215..d7378373a 100755
--- a/nexfetch
+++ b/nexfetch
@@ -1155,6 +1155,7 @@ get_distro() {
                 10.16*) codename="macOS Big Sur" ;;
                 11.*)  codename="macOS Big Sur" ;;
                 12.*)  codename="macOS Monterey" ;;
+                13.*)  codename="macOS Ventura" ;;
                 *)      codename=macOS ;;
             esac
 
