diff --git a/nexfetch b/nexfetch
index 48b96d215..0527482ef 100755
--- a/nexfetch
+++ b/nexfetch
@@ -783,7 +783,7 @@ image_source="auto"
 # NOTE: AIX, Hash, Alpine, AlterLinux, Amazon, Anarchy, Android, instantOS,
 #       Antergos, antiX, "AOSC OS", "AOSC OS/Retro", Apricity, ArchCraft,
 #       ArcoLinux, ArchBox, ARCHlabs, ArchStrike, XFerience, ArchMerge, Arch,
-#       Artix, Arya, Bedrock, Bitrig, BlackArch, BLAG, BlankOn, BlueLight,
+#       Artix, Arya, Aster, Bedrock, Bitrig, BlackArch, BLAG, BlankOn, BlueLight,
 #       Bodhi, bonsai, BSD, BunsenLabs, Calculate, Carbs, CentOS, Chakra, ChaletOS,
 #       Chapeau, Chrom*, Cleanjaro, ClearOS, Clear_Linux, Clover, Condres,
 #       Container_Linux, Crystal Linux, CRUX, Cucumber, dahlia, Debian, Deepin,
@@ -5142,7 +5142,7 @@ ASCII:
                                 NOTE: AIX, Hash, Alpine, AlterLinux, Amazon, Anarchy, Android,
                                 instantOS, Antergos, antiX, \"AOSC OS\", \"AOSC OS/Retro\",
                                 Apricity, ArchCraft, ArcoLinux, ArchBox, ARCHlabs, ArchStrike,
-                                XFerience, ArchMerge, Arch, Artix, Arya, Bedrock, Bitrig,
+                                XFerience, ArchMerge, Arch, Artix, Arya, Aster, Bedrock, Bitrig,
                                 BlackArch, BLAG, BlankOn, BlueLight, Bodhi, bonsai, BSD, BunsenLabs,
                                 Calculate, Carbs, CentOS, Chakra, ChaletOS, Chapeau, Chrom,
                                 Cleanjaro, ClearOS, Clear_Linux, Clover, Condres, Container_Linux,
@@ -6250,6 +6250,30 @@ ${c4}                    %%%
 EOF
         ;;
 
+        "Aster"*)
+            set_colors 6 6
+            read -rd '' ascii_data <<'EOF'                                        
+${c1}                      ...''...         
+${c1}                 .;oOXWMWNXXXNMMN0d:.   
+${c1}              .oXMWOo;..       ..:oO;   
+${c1}            ;KMWx,       co,            
+${c1}          'KMNl         dMMW.           
+${c1}         oMMx          xMMMMk           
+${c1}        xMM:          dMMMMMM;          
+${c1}       cMMl          dMMMMMMMW          
+${c1}       NMK          xMMMx::dXMx         
+${c1}      ,MMl         xMMN'     .o.        
+${c1}      cMM;        dMMW'                 
+${c1}      ;MMc       oMMW,                  
+${c1}       WMK      dMMW,  ccccccc.         
+${c1}       lMMl    oMMM;   ooooooo.         
+${c1}        OMMc   ...                      
+${c1}         xMMx                           
+${c1}          ;XMN:                         
+${c1}            ,.                          
+EOF
+      ;;
+
         "Bedrock"*)
             set_colors 8 7
             read -rd '' ascii_data <<'EOF'
diff --git a/nexfetch.1 b/nexfetch.1
index b90b57817..e66d6a4c9 100644
--- a/nexfetch.1
+++ b/nexfetch.1
@@ -304,7 +304,7 @@ Which Distro's ascii art to print
 NOTE: AIX, Hash, Alpine, AlterLinux, Amazon, Anarchy, Android, instantOS,
 Antergos, antiX, "AOSC OS", "AOSC OS/Retro", Apricity, ArchCraft,
 ArcoLinux, ArchBox, ARCHlabs, ArchStrike, XFerience, ArchMerge, Arch,
-Artix, Arya, Bedrock, Bitrig, BlackArch, BLAG, BlankOn, BlueLight,
+Artix, Arya, Aster, Bedrock, Bitrig, BlackArch, BLAG, BlankOn, BlueLight,
 bonsai, BSD, BunsenLabs, Calculate, Carbs, CentOS, Chakra, ChaletOS,
 Chapeau, Chrom*, Cleanjaro, ClearOS, Clear_Linux, Clover, Condres,
 Container_Linux, CRUX, Cucumber, dahlia, Debian, Deepin, DesaOS,
